tools/eMMC/functions.sh

```diff
@@ -254,140 +278,166 @@ dd_bootloader () {
+    sed -i -e 's:uuid=:#uuid=:g' ${tmp_rootfs_dir}/boot/uEnv.txt
+    echo "uuid=${root_uuid}" >> ${tmp_rootfs_dir}/boot/uEnv.txt
 
-    message="UUID=${root_uuid}" ; broadcast
+    echo_broadcast "===> UUID=${root_uuid}"
     root_uuid="UUID=${root_uuid}"
   else
-    #really a failure...
+    #FIXME: really a failure...
     root_uuid="${source}p${media_rootfs}"
   fi
 
-  message="Generating: /etc/fstab" ; broadcast
-  echo "# /etc/fstab: static file system information." > /tmp/rootfs/etc/fstab
-  echo "#" >> /tmp/rootfs/etc/fstab
-  echo "${root_uuid}  /  ext4  noatime,errors=remount-ro  0  1" >> /tmp/rootfs/etc/fstab
-  echo "debugfs  /sys/kernel/debug  debugfs  defaults  0  0" >> /tmp/rootfs/etc/fstab
-  cat /tmp/rootfs/etc/fstab
-
-  message="/boot/uEnv.txt: disabling eMMC flasher script" ; broadcast
-  sed -i -e 's:'$emmcscript':#'$emmcscript':g' /tmp/rootfs/boot/uEnv.txt
-  cat /tmp/rootfs/boot/uEnv.txt
-  message="-----------------------------" ; broadcast
-
+  echo_broadcast "==> Generating: /etc/fstab"
+  echo "# /etc/fstab: static file system information." > ${tmp_rootfs_dir}/etc/fstab
+  echo "#" >> ${tmp_rootfs_dir}/etc/fstab
+  echo "${root_uuid}  /  ext4  noatime,errors=remount-ro  0  1" >> ${tmp_rootfs_dir}/etc/fstab
+  echo "debugfs  /sys/kernel/debug  debugfs  defaults  0  0" >> ${tmp_rootfs_dir}/etc/fstab
+  cat ${tmp_rootfs_dir}/etc/fstab
+
+  #FIXME: What about when you boot from a fat partition /boot ?
+  echo_broadcast "==> /boot/uEnv.txt: disabling eMMC flasher script"
+  sed -i -e 's:'$emmcscript':#'$emmcscript':g' ${tmp_rootfs_dir}/boot/uEnv.txt
+  cat ${tmp_rootfs_dir}/boot/uEnv.txt
   flush_cache
-  umount /tmp/rootfs/ || umount -l /tmp/rootfs/ || write_failure
+
+  echo_broadcast "==> Unmounting ${tmp_rootfs_dir}"
+  umount ${tmp_rootfs_dir} || umount -l ${tmp_rootfs_dir} || write_failure
 
   if [ "x${is_bbb}" = "xenable" ] ; then
+    echo_broadcast "==> Stop Cylon LEDs"
     [ -e /proc/$CYLON_PID ]  && kill $CYLON_PID
   fi
 
-  message="Syncing: ${destination}" ; broadcast
+  echo_broadcast "==> Syncing: ${destination}"
   #https://github.com/beagleboard/meta-beagleboard/blob/master/contrib/bone-flash-tool/emmc.sh#L158-L159
   # force writeback of eMMC buffers
   sync
   dd if=${destination} of=/dev/null count=100000
-  message="Syncing: ${destination} complete" ; broadcast
-  message="-----------------------------" ; broadcast
+  echo_broadcast "Syncing: ${destination} complete"
+  echo_broadcast "-------------------------------------------------------------------"
 
   if [ -f /boot/debug.txt ] ; then
-    message="This script has now completed its task" ; broadcast
-    message="-----------------------------" ; broadcast
-    message="debug: enabled" ; broadcast
+    echo_broadcast "This script has now completed its task"
+    echo_broadcast "-----------------------------"
+    echo_broadcast "debug: enabled"
     inf_loop
   else
+    #FIXME: Why are we unmounting the host /tmp ?
```

Comment:
Yeah this is an old comment. In the rework code I figured why it was there and added teardown function that take care of that.